### DISCUSSION TITLE
Couldn't compile CGNS Version 4.3.0

### DISCUSSION THREAD

**Discussion by krendyasevdi** [CGNS/CGNS#341] [CGNStalk]
Hi, everyone!

I want to use CGNS Viewer for postprocessing some results, but I've met some difficulties with compiling.
Could you please help me understand the errors I've got after my compile attempt? Please see the more detailed information below.

# Attempt 1 – Windows
- configure.bat
- windows cmake
- windows make
	- Makefile:3: *** missing separator.  Stop.

# Attempt 2 – MinGW/MSYS2
- configure
- cmake
- make
```log
[ 18%] Building C object CMakeFiles/cgns_static.dir/cg_hashmap.o
/c/Users/dbobirev/Downloads/Syncs/OneDrive/Рабочий стол/CGNS-4.3.0/src/cg_hashmap.c: In function ‘cgi_calculate_keysize’:
/c/Users/dbobirev/Downloads/Syncs/OneDrive/Рабочий стол/CGNS-4.3.0/src/cg_hashmap.c:167:17: error: operator '==' has no left operand
  167 | #if SIZEOF_LONG == SIZEOF_VOID_P
      |                 ^~
make[2]: *** [CMakeFiles/cgns_static.dir/build.make:132: CMakeFiles/cgns_static.dir/cg_hashmap.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:110: CMakeFiles/cgns_static.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

# Setup
1. add tkWinInt.h to `C:\tools\Anaconda3\Library\include\win`
2. download the source .zip [here](https://github.com/CGNS/CGNS/releases/tag/v4.3.0) 
3. cd src 
4. configure.bat
5. (option) `set PATH=%PATH%;C:\Program Files\CMake\bin`
6. cmake .
7. if errors, comment out those lines (doesn’t seem essential)
8. make

**Comment by MicK7** [comment]
Have you already checked https://cgns.github.io/cgns-modern.github.io/general/FAQ.html#supportfaq ?

**Reply by krendyasevdi** [reply]
Thanks! Managed to build the library, but when I try to build cgnsview, it's trying to find a file cgns_static.lib, and it is not there.

Initially it would not build at all, but after mixing some files, I've managed to get this far.



I only have cgns.lib in my build_cgns\src\Debug, nothing called cgns_static. Maybe I've missed some build steps?

**Reply by Peter-P779** [reply]
I get the exact same error. What was the problem?

**Comment by krendyasevdi** [comment]
@MicK7, thanks! Just duplicating my previous reply here for greater user reach, because the problem is pretty much urgent.

Managed to build the library, but when I try to build cgnsview, it's trying to find a file cgns_static.lib, and it is not there.

Initially it would not build at all, but after mixing some files, I've managed to get this far.

I only have cgns.lib in my build_cgns\src\Debug, nothing called cgns_static. Maybe I've missed some build steps?

**Reply by MicK7** [reply]
Are you sure to have set the option `-D CGNS_USE_SHARED:BOOL=OFF` ?

**Reply by MicK7** [reply]
Maybe you can try with `-D CGNS_BUILD_SHARED:BOOL=ON -D CGNS_USE_SHARED:BOOL=ON`

**Reply by Musickiller** [reply]
Hi! Is there a need to build cgnsviewer at all? I after the CGNS itself is successfully built (exe files for tools are built too), there is still no exe files for gcnstools dir.

The bat file in the sources requires cgiowish.exe, which is not built together with CGNS itself

> `-D CGNS_BUILD_SHARED:BOOL=ON -D CGNS_USE_SHARED:BOOL=ON`

I've tried with both "ON" and "OFF" now

**Comment by krendyasevdi** [comment]
@MicK7, please see the @Musickiller reply. We would be much obliged if you'd help us overcome this trouble

**Reply by MicK7** [reply]
Since you seem to be using Anaconda for building CGNS, did you look at the recipee from conda-forge https://github.com/conda-forge/cgns-feedstock/blob/main/recipe/ ?
I don't know if cgnstools/cgnsview is tested on windows.
